firebase-perf/src/main/java/com/google/firebase/perf/internal/GaugeMetadataManager.java

```diff
@@ -40,22 +41,21 @@
 
   private static final AndroidLogger logger = AndroidLogger.getInstance();
 
+  private final Context applicationContext;
   private final Runtime runtime;
   private final ActivityManager activityManager;
+
   private final MemoryInfo memoryInfo;
   private final String currentProcessName;
-  private final Context applicationContext;
-
-  GaugeMetadataManager(Context applicationContext) {
-    this(Runtime.getRuntime(), applicationContext);
-  }
 
   @VisibleForTesting
-  GaugeMetadataManager(Runtime runtime, Context applicationContext) {
-    this.runtime = runtime;
+  @Inject
+  GaugeMetadataManager(
+      Runtime runtime, Context applicationContext, ActivityManager activityManager) {
     this.applicationContext = applicationContext;
```

Comment:
Why do you need to add `ActivityManager activityManager` in the constructor? Can you use `applicationContext.getSystemService(Context.ACTIVITY_SERVICE)` to get the activityManager?

Reply:
As part of the DI changes I am also looking into what dependencies we were previously instantiating within the class itself and should actually be injected from outside. This is one of those.

Reply:
I see. In the test, you may want to mock `activityManager` first and then pass it to `GaugeMetadataManager`'s constructor. The current test does not seem to benefit from your constructor change.

Reply:
It is being [shadowed](http://robolectric.org/extending/) instead of mocked as we can't mock some Android specific API's directly:

https://github.com/firebase/firebase-android-sdk/blob/ad95b179508f9246dba768786f2a85154ec9b9d1/firebase-perf/src/test/java/com/google/firebase/perf/internal/GaugeMetadataManagerTest.java#L77-L78

https://github.com/firebase/firebase-android-sdk/blob/ad95b179508f9246dba768786f2a85154ec9b9d1/firebase-perf/src/test/java/com/google/firebase/perf/internal/GaugeMetadataManagerTest.java#L91

---

FYI: I am now passing [spy](https://javadoc.io/static/org.mockito/mockito-core/3.8.0/org/mockito/Mockito.html#13) instead of actual objects to make our mocks/fakes clear.

Reply:
Thanks for the clarification on spy! My point was that you passed the real object instead of the spy one 
https://github.com/firebase/firebase-android-sdk/blob/ad95b179508f9246dba768786f2a85154ec9b9d1/firebase-perf/src/test/java/com/google/firebase/perf/internal/GaugeMetadataManagerTest.java#L66
so that you are not actually benefiting from making `activityManager` as injected.
Now I am seeing that you are passing the spy object instead of the real obejct here https://github.com/firebase/firebase-android-sdk/blob/a1b4308439ca1b40a09e4ec527ea6a0e7d8d114e/firebase-perf/src/test/java/com/google/firebase/perf/internal/GaugeMetadataManagerTest.java#L72-L75.
This is exactly what I was suggesting.